File: ElasticRead.js
const { Client } = require('@elastic/elasticsearch');

// Replace these values with your actual Elasticsearch connection details
const clusterId = 'dbe766fc301b4286b2b74ad8ba303fce';
const username = 'elastic';
const password = 'uQFTxSsR7Kmh72PAGM5mplDv';

// Create an Elasticsearch client with user access
const client = new Client({
  cloud: {
    id: clusterId,
  },
  auth: {
    username,
    password,
  },
});

// Example: Retrieve data from an index
const indexName = 'your_index_name';
const queryBody = {
  query: {
    match_all: {}  // You can customize your query here
  }
};

// Perform a search
async function searchData() {
  const { body } = await client.search({
    index: indexName,
    body: queryBody
  });

  // Process the result
  body.hits.hits.forEach(hit => {
    console.log(hit._source);
  });
}

// Call the function to perform the search
searchData();

File: ElasticRead.py
from elasticsearch import Elasticsearch
#requires Elasticsearch module - pip install elasticsearch 
# Replace 'your_connection_string' with your actual Elasticsearch connection string
connection_string = 'your_connection_string'

# Create an Elasticsearch client
es = Elasticsearch(connection_string)

# Example: Retrieve data from an index
index_name = 'your_index_name'
query_body = {
    "query": {
        "match_all": {}  # You can customize your query here
    }
}

# Perform a search
result = es.search(index=index_name, body=query_body)

# Process the result
for hit in result['hits']['hits']:
    print(hit['_source'])
